python/pyspark/sql/tests.py

```diff
@@ -459,6 +459,23 @@ def filename(path):
         row2 = df2.select(sameText(df2['file'])).first()
         self.assertTrue(row2[0].find("people.json") != -1)
 
+    def test_udf_defers_judf_initalization(self):
```

Comment:
I thought about it but I have this impression, maybe incorrect, that we avoid creating new contexts to keep total execution time manageable. If you think this justifies a separate  `TestCase` I am more than fine with that ([SPARK-19224](https://issues.apache.org/jira/browse/SPARK-19224) and [[PYSPARK] Python tests organization ](http://apache-spark-developers-list.1001551.n3.nabble.com/PYSPARK-Python-tests-organization-td20542.html), right?).

If not, we could mock this, and put assert on the number of calls.